cmd/loraserver/cmd/configfile.go

```diff
@@ -190,8 +190,20 @@ get_downlink_data_delay="{{ .NetworkServer.GetDownlinkDataDelay }}"
   # 0: RX1, fallback to RX2 (on RX1 scheduling error)
   # 1: RX1 only
   # 2: RX2 only
+  # 3: Same as 0, but dynamic decision if RX1 will be sent, see RX 2 only DataRate
```

Comment:
Note that changing the order in which `setTXInfoForRX1` and `setTXInfoForRX2` are being called, would already be sufficient I think.

Reply:
Thanks for the review. 

"Instead of using just RX2, I think it would be better to implement this in such a way that when RX2 is preferred, this turns this slice if frames into [Frame RX2, Frame RX1]. In this case when RX2 fails, there is still an option to re-try using RX1."

Does that mean, that even if RX2 fails, there is a chance for RX1 to succeed? That is not really intuitive to me, could you explain a bit more in detail on how this mechanism works (although I agree, that just changing the order of RX2 and RX1 calls would do the job)?

Reply:
A gateway might have something already in the queue at the RX2 timing causing the RX2 scheduling to fail. The gateway will then send a nACK to the NS causing the NS to retry by scheduling the downlink using RX1 parameters. In this case the RX1 might not collide with the other frame(s) in the gateway queue. Of course this only works when there is a low latency between the GW and NS.